packages/persist/lib/controllers/persist.controller.ts

```diff
@@ -126,8 +149,18 @@ class PersistController {
             params: { environmentId, nangoConnectionId, syncId, syncJobId },
             body: { model, records, providerConfigKey, connectionId, activityLogId }
         } = req;
-        const persist = async (dataRecords: DataRecord[]) => {
-            return await dataService.update(dataRecords, nangoConnectionId, model, activityLogId, environmentId);
+        const persist = async (records: FormattedRecord[], legacyRecords: DataRecord[]) => {
+            recordsService
+                .update(records, nangoConnectionId, model)
+                .then((res) => {
+                    if (isErr(res)) {
+                        throw res.err;
+                    }
+                })
+                .catch((reason) => {
+                    logger.error(`Failed to update records: ${reason}`);
+                });
```

Comment:
imo we should be consistent between async and then/catch, it's confusing to mix the syntax here

Reply:
that's only temporary for the double write. Once migration is done, I will cleanup the code and stick with `await`